### ISSUE TITLE
PPRMOCMagnesSDK arm64e

### ISSUE THREAD

**Issue by Nosskirneh** [braintree/braintree_ios#457]
The static library PPRMOCMagnesSDK is not compiled for arm64e. Is that something that will be released in the the near future?

**Comment by demerino** [comment]
@Nosskirneh Could share steps to reproduce this issue?

**Comment by Nosskirneh** [comment]
@demerino Sure!

First I should say that I don't use Xcode to compile my project, as I don't play inside of Apple's walled garden. I'm using [theos](https://github.com/theos/theos) to build. Anyway, I tried to compile my project with the Braintree static library being linked.

I then got this
```
==> Linking bundle NextUpPrefs (arm64)…
Undefined symbols for architecture arm64:
  "_OBJC_CLASS_$_PPRMOCMagnesSDK", referenced from:
      objc-class-ref in libBraintree.a(PPDataCollector.o)
ld: symbol(s) not found for architecture arm64
```

This has nothing to do with arm64e, but it cannot find the class `PPMocMagnesSDK` at all.

So I started digging and suspected that I needed to link the `libPPRiskMagnesOC.a` static library too. Said and done, this is what I get:
```
...
==> Linking bundle NextUpPrefs (arm64)…
==> Generating debug symbols for NextUpPrefs…
rm /Users/andeh/Dropbox/Developer/iOS/NextUp/.theos/obj/debug/arm64/NextUpRootListController.xm.mm
==> Preprocessing NextUpRootListController.xm…
==> Compiling NextUpRootListController.xm (arm64e)…
==> Linking bundle NextUpPrefs (arm64e)…
ld: warning: ignoring file /var/theos/lib/libPPRiskMagnesOC.a, missing required architecture arm64e in file /var/theos/lib/libPPRiskMagnesOC.a (5 slices)
Undefined symbols for architecture arm64e:
  "_OBJC_CLASS_$_KDataCollector", referenced from:
      objc-class-ref in libBraintree.a(BTDataCollector.o)
  "_OBJC_CLASS_$_PPRMOCMagnesSDK", referenced from:
      objc-class-ref in libBraintree.a(PPDataCollector.o)
ld: symbol(s) not found for architecture arm64e
clang: error: linker command failed with exit code 1 (use -v to see invocation)
```

It compiles fine for arm64 now (as the class is found), but not for arm64e since it seems like `PPRiskMagnesOC` is not compiled for it.

Maybe I'm doing something wrong. Thanks for helping out!

**Comment by demerino** [comment]
We've contacted the team responsible for `PPRiskMagnesOC` and will try to get a new version with arm64e support. 

I'm not familiar with Theos but looking at the dev wiki, it appears you might be able to work around the issue temporarily by omitting `arm64e` from the `ARCHS`: http://iphonedevwiki.net/index.php/Theos#Theos_variables

**Comment by Nosskirneh** [comment]
Thank you for the response!

I was able to temporary modify the Braintree library so that it would not use parts from that library. I did not know arm64e devices in most cases would run arm64 binaries, but that doesn't really matter now – after integrating Braintree in both the app and in the backend, it turns out that my application for using Braintree's API got denied. Apparently selling smaller software yields a "high risk", even though I have over 500 purchases and only one refund request since December. I'm really disappointed in both PayPal and Braintree (as PayPal discontinued their SDK in favor of their new acquisition).

All you ever see online is "it's so easy to integrate Braintree" – well, for what benefit if you're a small indie dev and not a gigantic company?

/rant over
Thanks for your help anyway
